### PR TITLE
Adds clustering data to the info returned about BQ tables

### PR DESCRIPTION
**Pull request by abair34**
While trying to use the API to return back the list of clustering columns for a table I noticed BigQueryTable.Resource.Clustering was empty.  It looks as if it was not being mapped to the returned data from the API call although the Clustering property existed in the table class.  

This commit added a unit test and populates the table's clustering info.

### CODE REVIEW

**Comment by abair34** [comment]
@googlebot I signed it!

**Comment by jskeet** [comment]
Thanks for this. It looks like there are a few other properties exposed in the `TableList.TablesData` that would be worth copying. I'm happy either for you to amend this PR, or we (@amanda-tarafa or I) could do it all in a new PR - up to you!

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Tests/BigQueryClientImplTest.cs [hunk 1]
```diff
@@ -597,8 +597,7 @@ public void CreateJob_Labels()
         [Fact]
         public void ListTablesConversion()
         {
-            var listResource = new TableList.TablesData
-            {
+            var listResource = new TableList.TablesData {
```
Comment (jskeet) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Tests/BigQueryClientImplTest.cs]: Please revert this change - it was fine as it was before.

Comment (abair34) [reply apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Tests/BigQueryClientImplTest.cs]: Ahh good eye!  I've got my VS formatting options set to put open {'s on the same line (probably not in alignment with the project).  I'll make sure to update them.  I'm new to contributing on github in general so I appreciate the help!  Thanks @jskeet!

Comment (jskeet) [reply apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Tests/BigQueryClientImplTest.cs]: Let me know if you need a hand with the git aspects of it. Once you've created an extra commit for this bit, if you use `git rebase -i master` then change the second and third lines to have "f" at the start instead of "pick", it will squash them all together. Then just run `git push -f` to update the PR.

**Comment by amanda-tarafa** [comment]
This has now been released in [Google.Cloud.BigQuery.V2 version 1.4.0-beta07](https://www.nuget.org/packages/Google.Cloud.BigQuery.V2/1.4.0-beta07).
Thanks for the PR!
